packages/grafana-ui/src/components/uPlot/config/UPlotAxisBuilder.ts

```diff
@@ -266,6 +225,60 @@ export function formatTime(
   return splits.map((v) => (v == null ? '' : dateTimeFormat(v, { format, timeZone })));
 }
 
+/* Minimum grid & tick spacing in CSS pixels */
+function calculateSpace(
+  self: uPlot,
+  axisIdx: number,
+  scaleMin: number,
+  scaleMax: number,
+  plotDim: number,
+  formatValue?: (value: unknown) => string
+): number {
+  const axis = self.axes[axisIdx];
+  const scale = self.scales[axis.scale!];
+
+  // for axis left & right
+  if (axis.side !== 2 || !scale) {
+    return plotDim <= Y_TICK_SPACING_PANEL_HEIGHT ? Y_TICK_SPACING_SMALL : Y_TICK_SPACING_NORMAL;
+  }
+
+  const maxTicks = plotDim / X_TICK_SPACING_NORMAL;
+  const increment = (scaleMax - scaleMin) / maxTicks;
+
+  let sample = '';
+
+  if (scale.time) {
+    sample = formatTime(self, [scaleMin], axisIdx, X_TICK_SPACING_NORMAL, increment)[0];
+  } else if (formatValue != null) {
+    sample = formatValue(scaleMin);
```

Comment:
probably missing the decimals arg handling here...

Reply:
err, no, we we dont have the tick increment here yet. the whole point of this function is to determine that increment :)